File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerines):
    dict = {}
    answer = 0
    for tangerine in tangerines:
        if tangerine not in dict:
            dict[tangerine] = 1
        else:
            dict[tangerine] += 1
    value = list(dict.values())
    value.sort(reverse=True)
    
    for val in value:
        k = k - val
        answer += 1
        if k <= 0:
            break
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Long> sizeFrequencyMap = Arrays.stream(tangerine)
            .boxed()
            .collect(Collectors.groupingBy(size -> size, Collectors.counting()));

        List<Map.Entry<Integer, Long>> sortedEntries = sizeFrequencyMap.entrySet().stream()
            .sorted((e1, e2) -> e2.getValue().compareTo(e1.getValue()))
            .collect(Collectors.toList());

        int varietyCount = 0;
        for (Map.Entry<Integer, Long> entry : sortedEntries) {
            k -= entry.getValue();
            varietyCount++;
            if (k <= 0) break;
        }

        return varietyCount;
    }
}
